### PR TITLE
[Release-5.1] Keep Updating coinbase during syncup of DS Node

### PR DESCRIPTION
**Pull request by sandipbhoir**
## Description
Keep Updating coinbase during syncup of DS Node to avoid failure ( `Failed to verify the Byzantine nodes to be removed`) at next DS block consensus.
Issue# - https://github.com/Zilliqa/Issues/issues/601

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libLookup/Lookup.cpp [hunk 1]
```diff
@@ -2057,12 +2057,36 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
...
+      m_mediator.m_ds->SaveCoinbase(txBlock.GetB1(), txBlock.GetB2(),
+                                    CoinbaseReward::FINALBLOCK_REWARD,
+                                    blockNum + 1);
+      // Need if it join immediately before vacaous. And will be used in
+      // InitCoinbase in final blk consensus in vacaous epoch.
+      m_mediator.m_ds->m_totalTxnFees += rewards;
+
+      if (!BlockStorage::GetBlockStorage().PutEpochFin(blockNum)) {
+        LOG_GENERAL(WARNING, "BlockStorage::PutEpochFin failed "
+                                 << m_mediator.m_currentEpochNum);
```
Comment (ansnunez) [inline src/libLookup/Lookup.cpp]: Should we use `blockNum + 1` instead of `m_currentEpochNum`?

File: src/libLookup/Lookup.cpp [hunk 2]
```diff
@@ -2057,12 +2057,36 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
...
+      LOG_GENERAL(INFO, "Update coin base for finalblock with blockNum: "
+                            << blockNum << ", reward: " << rewards);
+      m_mediator.m_ds->SaveCoinbase(txBlock.GetB1(), txBlock.GetB2(),
+                                    CoinbaseReward::FINALBLOCK_REWARD,
+                                    blockNum + 1);
+      // Need if it join immediately before vacaous. And will be used in
+      // InitCoinbase in final blk consensus in vacaous epoch.
+      m_mediator.m_ds->m_totalTxnFees += rewards;
+
+      if (!BlockStorage::GetBlockStorage().PutEpochFin(blockNum)) {
```
Comment (ansnunez) [inline src/libLookup/Lookup.cpp]: `blockNum` or `blockNum + 1`?

File: src/libLookup/Lookup.cpp (hunk 1 above)
Comment (sandipbhoir) [reply src/libLookup/Lookup.cpp]: Agree. Fixed

File: src/libLookup/Lookup.cpp [hunk 3]
```diff
@@ -2057,12 +2057,36 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
...
+      LOG_GENERAL(INFO, "Update coin base for finalblock with blockNum: "
+                            << blockNum << ", reward: " << rewards);
+      m_mediator.m_ds->SaveCoinbase(txBlock.GetB1(), txBlock.GetB2(),
+                                    CoinbaseReward::FINALBLOCK_REWARD,
+                                    blockNum + 1);
+      // Need if it join immediately before vacaous. And will be used in
+      // InitCoinbase in final blk consensus in vacaous epoch.
+      m_mediator.m_ds->m_totalTxnFees += rewards;
+
+      if (!BlockStorage::GetBlockStorage().PutEpochFin(blockNum + 1)) {
```
Comment (kaikawaliu) [inline src/libLookup/Lookup.cpp]: Seems PutEpochFin is not necessary here, can remove it for simplicity

**Pull request merged**